### PR TITLE
Refactor and log

### PR DESCRIPTION
**Pull request by dbrajovic**
CHAIN-56

Features the following:
- less logs in general, especially `debug`
- `Relayer` loop scans eth blocks once instead of twice (if both batch and valset relaying is enabled)
- `Oracle` loop is more robust to errors encountered during its lifecycle
- connection to load-balanced endpoints (tested)
- `relayer` mode (`BatchRequester` and `Relayer` loops) runs exclusively for ETH addresses that haven't been registered with an orchestrator previously. Upon a successful `MsgSetOrchestratorAddresses` peggo will always run in `validator` mode if configured with the registered ETH address

## Summary by CodeRabbit

- **New Features**
  - Introduced functionality for creating token batches in the orchestrator system.
  - Enhanced Ethereum event observation and processing with a new `oracle` struct.
  
- **Improvements**
  - Refined error handling and logging for better feedback during initialization and runtime.
  - Improved network and keyring configuration setup for Cosmos and Ethereum.

- **Updates**
  - Updated HTTP endpoint description from "coingecko api" to "pricefeed api."

- **Refactor**
  - Significant refactoring of key management and orchestration components for better maintainability and performance.

### CODE REVIEW

File: orchestrator/signer.go [hunk 1]
```diff
@@ -4,161 +4,111 @@ import (
...
 }
 
-func (l *ethSignerLoop) getUnsignedBatch(ctx context.Context) (*types.OutgoingTxBatch, error) {
-	var oldestUnsignedBatch *types.OutgoingTxBatch
-	getOldestUnsignedBatchFn := func() (err error) {
-		// sign the last unsigned batch, TODO check if we already have signed this
-		oldestUnsignedBatch, err = l.inj.OldestUnsignedTransactionBatch(ctx)
-		if errors.Is(err, cosmos.ErrNotFound) || oldestUnsignedBatch == nil {
-			return nil
-		}
```
Comment (albertchon) [inline orchestrator/signer.go:131]: are these cases safe to omit? @dbrajovic

File: orchestrator/signer.go [hunk 2]
```diff
@@ -168,43 +118,23 @@ func (l *ethSignerLoop) signBatch(ctx context.Context, batch *types.OutgoingTxBa
 	return nil
 }
 
-func (l *ethSignerLoop) getUnsignedValsets(ctx context.Context) ([]*types.Valset, error) {
-	var oldestUnsignedValsets []*types.Valset
-	getOldestUnsignedValsetsFn := func() (err error) {
-		oldestUnsignedValsets, err = l.inj.OldestUnsignedValsets(ctx)
-		if errors.Is(err, cosmos.ErrNotFound) || oldestUnsignedValsets == nil {
-			return nil
-		}
```
Comment (albertchon) [inline orchestrator/signer.go:177]: same here @dbrajovic

File: orchestrator/signer.go (hunk 1 above)
Comment (dbrajovic) [reply orchestrator/signer.go:131]: The actual [peggy query](https://github.com/InjectiveLabs/injective-core/blob/317c891263837c0ec79fe214f1e83f364275e017/injective-chain/modules/peggy/keeper/grpc_query.go#L144) returns an error only if the address is malformed. So the `ErrNotFound` check never practically triggers. The check was used to ensure no error bubbles up (with `|| oldestUnsignedBatch == nil`) which we still uphold

File: orchestrator/signer.go (hunk 2 above)
Comment (dbrajovic) [reply orchestrator/signer.go:177]: Same [explanation](https://github.com/InjectiveLabs/peggo/pull/106#discussion_r1497398563) for this as well

**Pull request merged**